helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixDataAccessor.java

```diff
@@ -143,6 +147,7 @@ public boolean createMaintenance(MaintenanceSignal maintenanceSignal) {
     case EXTERNALVIEW:
       // check if bucketized
       if (value.getBucketSize() > 0) {
+        validateBucketizedEnabled(path, false /*isRead*/);
```

Comment:
I see that there are so many places that we have to validate this. It seems it is only using baseDataAccessor to read/write, can you help me understand why we don't put this check inside baseDataAccessor?

And we don't check `Map<String, ZNRecord> map = bucketizer.bucketize(lastCurState.getRecord());
` in `ParticipantManager`?

Reply:
Please correct me if I am wrong. 
I think we could use the existing bucketized node handling branch if added in ZKHelixDataAccessor for minion code change. Checking in baceAccessor might require more code change and branching. 
Also we need to touch multiple places -- ZkBaseDataAccessor.get/setData ZkBaseDataAccessor.get/setChildren() and ZkBaseDataAccessor.update().

Reply:
I agree with @pkuwm that this may not be scalable and adding it to baseDataAccessor. Future developers might not be aware of this function and nothing enforces them to add it, then there could be unprotected bucketized read/write.